provider/azure/environ.go

```diff
@@ -1151,6 +1231,16 @@ func (env *azureEnviron) deleteVirtualMachine(
 		vhdBlob := vhdContainer.Blob(vmName)
 		_, err := vhdBlob.DeleteIfExists(nil)
 		return errors.Annotate(err, "deleting OS VHD")
+	} else {
+		// Delete the managed OS disk.
+		logger.Debugf("- deleting OS disk (%s)", vmName)
+		resultCh, errCh := diskClient.Delete(env.resourceGroup, vmName, nil)
+		if result, err := <-resultCh, <-errCh; err != nil {
+			if isNotFoundResponse(result.Response) {
+				return errors.NotFoundf("disk %q", vmName)
```

Comment:
If we get not found when deleting something, should we just ignore the error, or is NotFound ignored by the caller?

Reply:
good catch, we should be ignoring not found errors here and in all the other deletes in this method